lib/kubeclient/common.rb

```diff
@@ -39,6 +39,12 @@ module ClientMixin
       'fieldSelector' => :field_selector
     }.freeze
 
+    DELETE_ARGUMENTS = {
+      'gracePeriodSeconds' => :grace_period_seconds,
+      'orphanDependents'   => :orphan_dependents,
```

Comment:
Yes, [I agree](https://github.com/abonas/kubeclient/pull/255#discussion_r125842680)
- I think we should still support search arguments we used to for compatibility
- I think we should pass a `params` hash to distinguish from other types of params
@abonas WDYT?